ip4defrag/defrag.go

```diff
@@ -188,15 +198,20 @@ type fragmentList struct {
 // See: http://www.sans.org/reading-room/whitepapers/detection/ip-fragment-reassembly-scapy-33969
 func (f *fragmentList) insert(in *layers.IPv4) (*layers.IPv4, error) {
 	// TODO: should keep a copy of *in in the list
-	// or not (ie the packet source is reliable) ?
+	// or not (ie the packet source is reliable) ? -> depends on Lazy / last packet
 	fragOffset := in.FragOffset * 8
 	if fragOffset >= f.Highest {
 		f.List.PushBack(in)
 	} else {
 		for e := f.List.Front(); e != nil; e = e.Next() {
 			frag, _ := e.Value.(*layers.IPv4)
-			if in.FragOffset <= frag.FragOffset {
-				debug.Printf("defrag: inserting frag %d before existing frag %d \n",
+			if in.FragOffset == frag.FragOffset {
```

Comment:
Line 121, add a call to the new flush method.

Well, you are right on the case you mention. It would imply to re-read the list to replace the first segment with the newer...